### PR TITLE
[Core] Defining Name methods for strategies

### PR DESCRIPTION
**Pull request by loumalouomega**
It is a static method. Part of #3185. The string corresponds with the name defined in the parameters, which does not necessarily coincide with the name of the class

### CODE REVIEW

File: kratos/solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h [hunk 1]
```diff
@@ -376,6 +376,15 @@ class ResidualBasedIncrementalUpdateStaticScheme
         this->mpDofUpdater->Clear();
     }
 
+    /**
+     * @brief Returns the name of the class as used in the settings (snake_case format)
+     * @return The name of the class
+     */
+    static std::string Name()
+    {
+        return "static";
```
Comment (philbucher) [inline kratos/solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h]: Why without „_scheme“?

**Comment by loumalouomega** [comment]
I considered the names we agreed in the other PR...

**Comment by philbucher** [comment]
Was there an agreement?
Now it is inconsistent

**Comment by philbucher** [comment]
But overall I think this is the way to go
=> we need to ensure that the names are unique, but I think this is done when they are being registered in the KratosComponents. What do you think?

**Comment by loumalouomega** [comment]
I don't know, for me the same name as the class in snake case will do the
trick

**Comment by philbucher** [comment]
my point is that you have:
block_**builder_and_solver**
newton_raphson_**strategy**
residual_**critaria**

so in order to be consistent it should be 
bossak_**scheme**

**Comment by loumalouomega** [comment]
OK

**Comment by loumalouomega** [comment]
Done @philbucher

**Comment by philbucher** [comment]
I will review this in combination with #3185 to see how the templates and static methods are working in combination

**Comment by loumalouomega** [comment]
@philbucher ?

**Comment by philbucher** [comment]
I thought this was no longer necessary with the latest version of #3185?

**Comment by loumalouomega** [comment]
> 
> 
> I thought this was no longer necessary with the latest version of #3185?

I can remove it, but is we want to consider an automatic macro, it is necessary

**Review by philbucher — APPROVED** [review]

**Pull request merged**